R/MAE_to_PSet.R

```diff
@@ -0,0 +1,152 @@
+#' Convert MultiAssayExperiment to TreatmentResponseExperiment
+#'
+#' This function converts a MultiAssayExperiment generated by gDR into a TreatmentResponseExperiment for use in the PharmacoGx package. The resulting PharmacoSet can be used for pharmacogenomic analysis of drug response.
+#'
+#' @param mae A MultiAssayExperiment object generated by gDR.
+#' @param pset_name A character string specifying the name of the resulting PharmacoSet object.
+#' 
+#' @return A PharmacoSet object.
+#' 
+#' @importFrom checkmate assert_class assert_string
+#' @importFrom data.table rbindlist setnames
+#' @importFrom gDRutils MAEpply convert_se_assay_to_dt
+#' @importFrom MultiAssayExperiment ExperimentList
+#' @importFrom SummarizedExperiment SummarizedExperiment rowData colData
+#' @importFrom CoreGx TreatmentResponseExperiment
+#' @importFrom PharmacoGx PharmacoSet2
+#' 
+#' @examples
+#' # Convert a MultiAssayExperiment object to a PharmacoSet object
+#' pset <- convert_MAE_to_PSet(mae, "my_pset")
+#' 
+#' @export
+convert_MAE_to_PSet <- function(mae,
+                                pset_name) {
+    checkmate::assert_class(mae, "MultiAssayExperiment")
+    checkmate::assert_string(pset_name)
+    # checkmate::assert_named(mae, "assays")
+    # checkmate::assert_named(mae, "colData")
+    # checkmate::assert_named(mae, "rowData")
+    # checkmate::assert_named(mae, "metadata")
+
+
+    # Get the unique assay names from all experiments combined
+    assay_names <- unique(unlist(gDRutils::MAEpply(mae, function(x) assayNames(x))))
+
+    # Create a list of data.tables, one for each assay name, with the data from all Summarized Experiments in the MAE. 
+    assay_list <- lapply(assay_names, function(x){
+        
+        # Get list of all the Summarized Experiments with all the unique Assays across the entire MAE
+        # If the summarized experiment does not have a given assay, create it 
+        se_list <- gDRutils::MAEpply(mae = mae, FUN = .extract_or_create_assay, unify=FALSE, x)
+        
+        # Get list of datatables with all assays with name=x converted to a data.table
+        se_dt_list <- lapply(se_list, FUN = gDRutils::convert_se_assay_to_dt, x)
+
+        # Combine all the data.tables into one data.table, add the name of the Summarized Experiment as a "data_type" column
+        # so that users can identify which Summarized Experiment the data came from
```

Comment:
```suggestion
        # Combine all the data.tables into one data.table, add the name of the Summarized Experiment
        # as a "data_type" column so that users can identify which Summarized Experiment the data came from
```